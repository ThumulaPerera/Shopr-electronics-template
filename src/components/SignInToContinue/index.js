import React from 'react'
import { Button, Header, Icon, Segment } from 'semantic-ui-react'

import FbSignInButton from '../FbSignInButton'

function SignInToContinue({ icon }) {
    return (
        <Segment placeholder>
            <Header icon>
                <Icon name={icon} />
                Sign in to view this page
            </Header>
            <FbSignInButton />
        </Segment>
    )
}

export default SignInToContinue
