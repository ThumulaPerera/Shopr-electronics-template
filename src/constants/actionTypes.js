export const AUTH_ACTION_TYPES = {
    LOGIN_SUCCESS : 'LOGIN_SUCCESS',
    LOGIN_ERROR : 'LOGIN_ERROR',
    SIGNOUT_SUCCESS : 'SIGNOUT_SUCCESS',
    SIGNOUT_ERROR : 'SIGNOUT_ERROR',
}

export const CART_ACTION_TYPES = {
    DELETE_IN_PROGRESS : 'DELETE_IN_PROGRESS',
    DELETE_SUCCESS : 'DELETE_SUCCESS',
    DELETE_ERROR : 'DELETE_ERROR',
    EDIT_IN_PROGRESS : 'EDIT_IN_PROGRESS',
    EDIT_SUCCESS : 'EDIT_SUCCESS',
    EDIT_ERROR : 'EDIT_ERROR',
}

