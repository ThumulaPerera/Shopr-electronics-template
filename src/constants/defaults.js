/* eslint-disable import/prefer-default-export */
export const defaultImgUrl = 'https://firebasestorage.googleapis.com/v0/b/ecom-cse.appspot.com/o/electronics-template%2Fdefault%20images%2FdefaultItem.jpg?alt=media&token=25d58086-a46e-42dc-9228-82a3b1a116d9';

export const browseProductsPageImgUrl = 'https://firebasestorage.googleapis.com/v0/b/ecom-cse.appspot.com/o/electronics-template%2Ftemplate%20screenshots%2Fbrowse%20products%20page.jpg?alt=media&token=1b13816e-ff59-4482-8301-e30d52a89162';
export const itemPageImgUrl = 'https://firebasestorage.googleapis.com/v0/b/ecom-cse.appspot.com/o/electronics-template%2Ftemplate%20screenshots%2Fitem%20page.jpg?alt=media&token=54839d86-7da3-419d-a5a8-19343289051b';
export const cartPageImgUrl = 'https://firebasestorage.googleapis.com/v0/b/ecom-cse.appspot.com/o/electronics-template%2Ftemplate%20screenshots%2Fcart%20page.jpg?alt=media&token=7cd2aed4-52a6-42a1-a154-a078e3f4861b';
export const myPurshasesPageImgUrl = 'https://firebasestorage.googleapis.com/v0/b/ecom-cse.appspot.com/o/electronics-template%2Ftemplate%20screenshots%2Forders%20page.jpg?alt=media&token=7293f018-0fbd-4eb7-8c47-af0ded528358';
