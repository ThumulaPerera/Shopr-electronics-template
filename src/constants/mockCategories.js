/* eslint-disable import/prefer-default-export */
export const categories = {
  1: {
    name: 'Computers and Laptops',
  },
  2: {
    name: 'Cameras and Accessories',
  },
  3: {
    name: 'Smart Phones and Tablets',
  },
  4: {
    name: 'TVs and Home Entertainmnt',
  },
  5: {
    name: 'Batteries and Power',
  },
};
