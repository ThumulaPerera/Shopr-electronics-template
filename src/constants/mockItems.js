export const items = {
    1: {
        name: "camera 1",
        imageURL: "https://d2yisiyyppbfr4.cloudfront.net/catalog/product/cache/image/e9c3970ab036de70892d86c6d221abfe/1/5/1519617660000_img_949161_1.jpg",
        description: "description of camera 1",
        currency: "$",
        price: "100",
        rating: 1
    },
    2: {
        name: "camera 2",
        imageURL: "https://www.sony-asia.com/image/f6fa6732135c962d2a79f89a9ff3ee6a?fmt=pjpeg&wid=1014&hei=396&bgcolor=F1F5F9&bgc=F1F5F9",
        description: "description of camera 2",
        currency: "$",
        price: "200",
        rating: 2
    },
    3: {
        name: "camera 3",
        imageURL: "https://d2yisiyyppbfr4.cloudfront.net/catalog/product/cache/image/e9c3970ab036de70892d86c6d221abfe/1/5/1519617660000_img_949161_1.jpg",
        description: "description of camera 3",
        currency: "$",
        price: "300",
        rating: 3
    },
    4: {
        name: "camera 4",
        imageURL: "https://d2yisiyyppbfr4.cloudfront.net/catalog/product/cache/image/e9c3970ab036de70892d86c6d221abfe/1/5/1519617660000_img_949161_1.jpg",
        description: "description of camera 4",
        currency: "$",
        price: "400",
        rating: 4
    },
    5: {
        name: "camera 1",
        imageURL: "https://d2yisiyyppbfr4.cloudfront.net/catalog/product/cache/image/e9c3970ab036de70892d86c6d221abfe/1/5/1519617660000_img_949161_1.jpg",
        description: "description of camera 1",
        currency: "$",
        price: "100",
        rating: 1
    },
    6: {
        name: "camera 2",
        imageURL: "https://www.sony-asia.com/image/f6fa6732135c962d2a79f89a9ff3ee6a?fmt=pjpeg&wid=1014&hei=396&bgcolor=F1F5F9&bgc=F1F5F9",
        description: "description of camera 2",
        currency: "$",
        price: "200",
        rating: 2
    },
    7: {
        name: "camera 3",
        imageURL: "https://d2yisiyyppbfr4.cloudfront.net/catalog/product/cache/image/e9c3970ab036de70892d86c6d221abfe/1/5/1519617660000_img_949161_1.jpg",
        description: "description of camera 3",
        currency: "$",
        price: "300",
        rating: 3
    },
    8: {
        name: "camera 4",
        imageURL: "https://d2yisiyyppbfr4.cloudfront.net/catalog/product/cache/image/e9c3970ab036de70892d86c6d221abfe/1/5/1519617660000_img_949161_1.jpg",
        description: "description of camera 4",
        currency: "$",
        price: "400",
        rating: 4
    },
    9: {
        name: "camera 3",
        imageURL: "https://d2yisiyyppbfr4.cloudfront.net/catalog/product/cache/image/e9c3970ab036de70892d86c6d221abfe/1/5/1519617660000_img_949161_1.jpg",
        description: "description of camera 3",
        currency: "$",
        price: "300",
        rating: 3
    },
    10: {
        name: "camera 4",
        imageURL: "https://d2yisiyyppbfr4.cloudfront.net/catalog/product/cache/image/e9c3970ab036de70892d86c6d221abfe/1/5/1519617660000_img_949161_1.jpg",
        description: "description of camera 4",
        currency: "$",
        price: "400",
        rating: 4
    },
    11: {
        name: "camera 1",
        imageURL: "https://d2yisiyyppbfr4.cloudfront.net/catalog/product/cache/image/e9c3970ab036de70892d86c6d221abfe/1/5/1519617660000_img_949161_1.jpg",
        description: "description of camera 1",
        currency: "$",
        price: "100",
        rating: 1
    },
    12: {
        name: "camera 2",
        imageURL: "https://www.sony-asia.com/image/f6fa6732135c962d2a79f89a9ff3ee6a?fmt=pjpeg&wid=1014&hei=396&bgcolor=F1F5F9&bgc=F1F5F9",
        description: "description of camera 2",
        currency: "$",
        price: "200",
        rating: 2
    },
    13: {
        name: "camera 3",
        imageURL: "https://d2yisiyyppbfr4.cloudfront.net/catalog/product/cache/image/e9c3970ab036de70892d86c6d221abfe/1/5/1519617660000_img_949161_1.jpg",
        description: "description of camera 3",
        currency: "$",
        price: "300",
        rating: 3
    },
    14: {
        name: "camera 4",
        imageURL: "https://d2yisiyyppbfr4.cloudfront.net/catalog/product/cache/image/e9c3970ab036de70892d86c6d221abfe/1/5/1519617660000_img_949161_1.jpg",
        description: "description of camera 4",
        currency: "$",
        price: "400",
        rating: 4
    },
    15: {
        name: "camera 1",
        imageURL: "https://d2yisiyyppbfr4.cloudfront.net/catalog/product/cache/image/e9c3970ab036de70892d86c6d221abfe/1/5/1519617660000_img_949161_1.jpg",
        description: "description of camera 1",
        currency: "$",
        price: "100",
        rating: 1
    },
    16: {
        name: "camera 2",
        imageURL: "https://www.sony-asia.com/image/f6fa6732135c962d2a79f89a9ff3ee6a?fmt=pjpeg&wid=1014&hei=396&bgcolor=F1F5F9&bgc=F1F5F9",
        description: "description of camera 2",
        currency: "$",
        price: "200",
        rating: 2
    },
    17: {
        name: "camera 3",
        imageURL: "https://d2yisiyyppbfr4.cloudfront.net/catalog/product/cache/image/e9c3970ab036de70892d86c6d221abfe/1/5/1519617660000_img_949161_1.jpg",
        description: "description of camera 3",
        currency: "$",
        price: "300",
        rating: 3
    },
    18: {
        name: "camera 4",
        imageURL: "https://d2yisiyyppbfr4.cloudfront.net/catalog/product/cache/image/e9c3970ab036de70892d86c6d221abfe/1/5/1519617660000_img_949161_1.jpg",
        description: "description of camera 4",
        currency: "$",
        price: "400",
        rating: 4
    },
    19: {
        name: "camera 3",
        imageURL: "https://d2yisiyyppbfr4.cloudfront.net/catalog/product/cache/image/e9c3970ab036de70892d86c6d221abfe/1/5/1519617660000_img_949161_1.jpg",
        description: "description of camera 3",
        currency: "$",
        price: "300",
        rating: 3
    },
    20: {
        name: "camera 4",
        imageURL: "https://d2yisiyyppbfr4.cloudfront.net/catalog/product/cache/image/e9c3970ab036de70892d86c6d221abfe/1/5/1519617660000_img_949161_1.jpg",
        description: "description of camera 4",
        currency: "$",
        price: "400",
        rating: 4
    },
}