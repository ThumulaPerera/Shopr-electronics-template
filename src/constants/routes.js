export const MAIN_ROUTE = '/:storeID';

export const HOME_ROUTE = '';
export const MY_PURCHASES_ROUTE = '/MyPurchases';
export const CART_ROUTE = '/Cart';
export const PROFILE_ROUTE = '/Profile';

