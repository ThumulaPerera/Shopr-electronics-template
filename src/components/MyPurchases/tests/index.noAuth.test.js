/* eslint-disable no-undef */
import React from 'react';
import { cleanup } from '@testing-library/react';
import { Route, MemoryRouter } from 'react-router-dom';
import { shallow } from 'enzyme';

import { MyPurchases as UnconnectedMyPurchases } from '../index';
import SignInToContinue from '../../SignInToContinue';

import { MAIN_ROUTE, MY_PURCHASES_ROUTE } from '../../../constants/routes';

jest.unmock('react-redux-firebase');
const rrf = require.requireActual('react-redux-firebase');

let component;

beforeEach(() => {
  // mock fn for isLoaded
  rrf.isLoaded = jest.fn(() => true);

  // eslint-disable-next-line no-console
  console.error = jest.fn(); /* disable proptype warnings */

  component = shallow(
    <MemoryRouter initialEntries={[{ pathname: `/storeId123${MY_PURCHASES_ROUTE}`, key: 'testKey' }]}>
      <Route path={`${MAIN_ROUTE}${MY_PURCHASES_ROUTE}`}>
        <UnconnectedMyPurchases
          auth={{}}
        />
      </Route>
    </MemoryRouter>
    ,
  ).find(UnconnectedMyPurchases).shallow();
});

afterEach(cleanup);

it('renders sign in to continue if auth.uid is absent', () => {
  expect(component).toMatchSnapshot();

  // renders sign in to continue
  expect(component.find(SignInToContinue).length).toEqual(1);
});
