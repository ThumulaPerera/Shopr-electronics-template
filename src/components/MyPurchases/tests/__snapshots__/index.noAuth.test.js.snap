// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders sign in to continue if auth.uid is absent 1`] = `
<SignInToContinue
  icon={null}
/>
`;
